code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -711,3 +711,57 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
 /mob/living/simple_animal/pet/dog/corgi/lisa/Life(delta_time = SSMOBS_DT, times_fired)
 	. = ..()
 	make_babies()
+
+/mob/living/simple_animal/pet/dog/breaddog //Most of the code originates from Cak
+	name = "Kobun"
+	desc = "It is a dog made out of bread. 'The universe is definitely half full'."
+	icon_state = "breaddog"
+	icon_living = "breaddog"
+	icon_dead = "breaddog_dead"
+	head_icon = 'icons/mob/pets_held_head.dmi'
+	health = 50
+	maxHealth = 50
+	gender = NEUTER
+	harm_intent_damage = 10
+	butcher_results = list(/obj/item/organ/internal/brain = 1, /obj/item/organ/internal/heart = 1, /obj/item/food/breadslice = 3,  \
+	/obj/item/food/meat/slab = 2)
+	response_harm_continuous = "takes a bite out of"
+	response_harm_simple = "take a bite out of"
+	attacked_sound = 'sound/items/eatfood.ogg'
+	held_state = "breaddog"
+	worn_slot_flags = ITEM_SLOT_HEAD
+
+/mob/living/simple_animal/pet/dog/breaddog/add_cell_sample()
+	return
+
+/mob/living/simple_animal/pet/dog/breaddog/CheckParts(list/parts)
+	..()
+	var/obj/item/organ/internal/brain/candidate = locate(/obj/item/organ/internal/brain) in contents
+	if(!candidate || !candidate.brainmob || !candidate.brainmob.mind)
+		return
+	candidate.brainmob.mind.transfer_to(src)
+	to_chat(src, "[span_boldbig("You are a bread dog!")]<b> You're a harmless dog/bread hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
+	so quickly that it generally doesn't matter. You're remarkably resilient to any damage besides this and it's hard for you to really die at all. You should go around and bring happiness and \
+	free bread to the station!</b><b>'I’m not alone, and you aren’t either'</b>")
+	var/default_name = "Kobun"
+	var/new_name = sanitize_name(reject_bad_text(tgui_input_text(src, "You are the [name]. Would you like to change your name to something else?", "Name change", default_name, MAX_NAME_LEN)))
+	if(new_name)
+		to_chat(src, span_notice("Your name is now <b>[new_name]</b>!"))
+		name = new_name
+
+/mob/living/simple_animal/pet/dog/breaddog/Life(delta_time = SSMOBS_DT, times_fired)
+	..()
+	if(stat)
+		return
+
+	if(health < maxHealth)
+		adjustBruteLoss(-4 * delta_time) //Fast life regen
+
+	for(var/mob/living/carbon/L in view(3, src)) //Mood aura which stay as long you do not wear Sanallite as hat or carry(I will try to make it work with hat someday(obviously weaker than normal one))
+		L.add_mood_event("kobun", /datum/mood_event/kobun)
```

Comment:
is there a proper name for this that isn't just L? what is this carbon to this proc?